format/view-spec.md

```diff
@@ -158,6 +174,57 @@ Each entry in `version-log` is a struct with the following fields:
 | _required_  | `timestamp-ms` | Timestamp when the view's `current-version-id` was updated (ms from epoch) |
 | _required_  | `version-id`   | ID that `current-version-id` was set to |
 
+#### Partial identifier
+
+The partial identifier holds a reference, containing a namespace and a name, of a table or view in the catalog.
+
+| Requirement | Field name     | Description |
+|-------------|----------------|-------------|
+| _required_  | `namespace`   | A list of namespace levels |
+| _required_  | `name`   | A string specifying the name of the source table |
+
+### Materialized View Metadata stored as part of the Table Metadata
+
+To be able to determine the freshness of the precomputed data, additional metadata is stored as part of the storage table.
+
+For that the additional field "refresh-state" is introduced as an opaque record in the table snapshot summary.
+
+| Requirement | Field name     | Description |
+|-------------|----------------|-------------|
+| _required_  | `refresh-state` | A [refresh state](#refresh-state) record stored as a JSON-encoded string | 
+
+#### Refresh state
+
+The refresh state record captures the state of all source tables and source views in the fully expanded query tree of the materialized view, including indirect references. It has the following fields:
+
+| Requirement | Field name     | Description |
+|-------------|----------------|-------------|
+| _required_  | `refresh-version-id` | The `version-id` of the materialized view when the refresh operation was performed  | 
+| _required_  | `source-table-states`   | A list of [source table](#soure-table) records for all tables that are directly or indirectly referenced in the materialized view query |
+| _required_  | `source-view-states`   | A list of [source view](#soure-view) records for all views that are directly or indirectly referenced in the materialized view query |
+| _required_  | `refresh-start-timestamp-ms` | A timestamp of when the refresh operation was started |
+
+#### Source table
+
+A source table record captures the state of a source table at the time of the last refresh operation.
+
+| Requirement | Field name     | Description |
+|-------------|----------------|-------------|
+| _required_  | `uuid` | The uuid of the source table | 
```

Comment:
Good point. We initially wanted to use identifiers here because of the point you mentioned. However, it turns out that identifiers might not be the same for different query engines because the catalog_name might be different. That's why it's problematic to include identifiers in the specification. 
The current workaround is to get the identifiers by pressing the SQL definition of the materialized view.

Reply:
@karuppayya There's no plan to have catalog APIs that looks up tables/views by UUID.  I don't think that's a good idea either.